### DISCUSSION TITLE
Chinese version test

### DISCUSSION THREAD

**Discussion by andrewpono** [ataradov/usb-sniffer#49] [General]
Out of curiosity I've purchased the Cn [version](https://www.aliexpress.us/item/3256807265784243.html?gatewayAdapt=glo2usa) of the sniffer from Ali sold by EVIT store. I've selected that store for several reasons:
1. Decent attribution to the original source.
2. Specific mention of the speed grade 6 FPGA. The other ones appear to use grade 4. Not sure how they pulled that off, 6 is pricey.
3. Acrylic "case" included.
There are many other offerings with varying degree of resemblance to the original, this one appears to have sustained minimal edits.
The board is exactly as the ad pics show, you get CBM9002A instead of the Cypress part, GSV332 comparators instead of Maxims.
The device comes with acrylic "case" already assembled and works out of the box.
I've dumped the EEPROM and compared it to the bin file from the repo. It matches, short of first 12 bytes and S/N at the very end. I then put the sniffer into the boot mode and tried to use the inf file to assign WinUSB driver to it. That did not work, Win errored out on it, despite VID/PID match. I had to use Zadig to assign the driver. After that I've loaded the bin file to RAM, no errors, then to  the EEPROM, again no problems. Dumped new EEPROM contents, compared to what was there as shipped, exact match.
Since there is no easy way to read the FPGA config, I did SRAM loading only, no flash. That worked fine, tool did not complain. The bitstream was converted from the repo's jed file. The sniffer kept sniffing, I assume the FPGA is either genuine or a very good clone. It looks genuine to me too, but it is hard to tell for sure these days.
The above leads me to believe both the original HW design and FW were left unchanged.
Speed test results:
Starting speed test
Transfer rate: 47.96 MB/s
Transfer rate: 47.49 MB/s
Transfer rate: 47.42 MB/s
Transfer rate: 47.81 MB/s
Generally, 47-48 MB/s, as promised.
I also did some signal integrity testing using PassMark Loopback [device](https://www.passmark.com/products/usb3loopback/index.php).
Attached are the results. Test fails consistently with mostly read timeouts when plugged into the sniffer, however if sniffer is not powered up (the host USB is disconnected), test shows results similar to direct connection. Not yet sure what to make of it.

![sniffer1](https://github.com/user-attachments/assets/34127c15-7c9f-4517-8afa-d577f5d1dde0)
[DirectHub.txt](https://github.com/user-attachments/files/18203833/DirectHub.txt)
[sniffer_failed3.txt](https://github.com/user-attachments/files/18203835/sniffer_failed3.txt)
[sniffer_noPower.txt](https://github.com/user-attachments/files/18203837/sniffer_noPower.txt)

**Comment by ataradov** [comment]
This is strange. Normally not powered sniffer will kill communication, since all the clamping circuits would not be powered. 

The other thing I would try is shorter cables. With pass-through you end up with double the cable length.

**Reply by andrewpono** [reply]
The cables are both 0.5m, one comes with the loopback gizmo, the other is by Tripp-Lite, sorta brand :)
It is all plugged into the Dell monitor USB3 hub for now. I might do some more testing later.
The main purpose of this writing was not to complain but to give others an idea of the Cn version details. If one can't or does not want to assemble, it's a viable option IMHO.

**Comment by ataradov** [comment]
I just ordered one to see how it compares on the same setup. It is good to see alternatives to Cypress ICs that are not afraid to market themselves without fudging the part numbers.

Cypress sales people actually reached out with the new IC they have based on the ARM core, which would be great. But it uses really stupid package, so not really useful.

**Reply by andrewpono** [reply]
I have a clone of the Lattice HW-USBN-2A JTAG programmer, it uses the same CBM9002A chip. Works very well, although the speed is not a big issue in this case I guess.
![Annotation 2024-12-19 152628](https://github.com/user-attachments/assets/72ac76a5-a948-4863-9baf-084a92b2f639)

**Comment by stappersg** [comment]
On Thu, Dec 19, 2024 at 03:14:03PM -0800, Andrew wrote:
> It is all plugged into the Dell monitor USB3 hub for now.

Oh

**Comment by andrewpono** [comment]
Well, FWIW, the same tests on the same HW are now passing just fine. The only change is the PC rebooted due to updates since then.

**Comment by stappersg** [comment]
On Fri, Dec 27, 2024 at 04:34:57PM -0800, Andrew wrote:
> The only change is the PC rebooted due to updates since then.

Please state which operating system was rebooted (and updated).


Regards
Geert Stappers
Happy Linux User
-- 
Silence is hard to parse

**Reply by andrewpono** [reply]
Win 10 Pro (22H2)

**Comment by ataradov** [comment]
I've got the hardware.  On the same system and in the same setup it works basically identically to mine. The included USB cables are not great. It seems to work for the host side, but for the target side they don't work with HS devices. They are likely marginal and additional distortion added  by the sniffer breaks it. Everything works with thick and actually standard compliant Amazon Basics cables. 

I also confirmed that neither sniffers kills communication when not powered, so this is expected, I guess.

Given that FPGA speed grade even higher than required, I would recommend this version. Unless until they optimize for cost as well. 

Also, FX2LP replacement IC used is this https://www.lcsc.com/product-detail/Microcontrollers-MCU-MPU-SOC_Corebai-Microelectronics-CBM9002A-56ISG_C476255.html None are in stock, but $3.9 is about how much the original should really cost. And commercial temperature range is even cheaper.

**Reply by andrewpono** [reply]
Good to get the creator's stamp of approval :)
Does the FPGA look genuine to you as well? I don't quite get the economics, given the real one is close to half of what they sell the board for.

**Comment by ataradov** [comment]
They are not that expensive. FPGAs get random markup because there are not easily available alternatives. Even LCSC lists them for $6 at 200 pcs quantity (not in stock at that time). And if you do contract assembly, you will have access to surplus at even cheaper prices. If you have a customer that needs 900 boards, you can add 100 devices for yourself and get them at a steep discount. 

But yes, it looks fine to me. There are no known fakes for those FPGAs, so the only possible suspicion would be re-marking slower grade. And I don't see any evidence of that.

**Comment by ataradov** [comment]
One more thing I just noticed about that board. They removed the trigger input and associated circuitry.  It is not a huge deal, but if you need it, you are not going to have it.

**Reply by andrewpono** [reply]
Yes, I saw that too. Puzzling, to me, why not leave a test point at the very least? But I figured it's a minor issue, it could have been much worse if they got really creative. There is another version where they added ESD suppressors to the USB lines, someone claimed it's incompatible with the original design, not sure it that's true however.

**Comment by jkblm** [comment]
I've purchased the CN sniffer last month from eBay.  It came with a nice steel case.
I used your great documentation posted here.  
First ran a speed test and received good results like you posted 47-48MB/sec.
Ran: sudo dmesg  ==> and reports back your name and a long serial #.
So I assumed it was programmed and ready to go (no documentation) in the package.
Set things up for Wireshark as you documented. It runs great with host Ubuntu 24.04

FPGA re-programs well and reports back - Programmed.
But both MCU SRAM and eeprom programming steps report: Error: Could not open un configured FX2LP device.

I open the case to check the components.  All look like what you have listed on the BOM but the Cypress MCU is 
a CBM9002A-56ISG.  Does the two files for MCU and eeprom work on the non-Cypress device?
Should I try to purchase an EOL Cypress and replace the CN device?

Thank you for your support on this USB Sniffer.  Jeff
![2020_0313_094607_011](https://github.com/user-attachments/assets/41d76398-b953-44ed-b51e-680a2a4aa422)

**Comment by ataradov** [comment]
All alternative devices are compatible, there is no need to replace anything.

SRAM programming only works on a EEPROM that was not previously programmed. This is a bootstrapping from zero step.

If you want to see it work, short BOOT test points (next to EEPROM) and re-plug USB. It will cause it to fail reading EEPROM, so it will boot into ROM mode. But there is no need to do that if device already enumerates as a USB sniffer.

EEPROM reprogramming should work, it does not look for FX2LP VID/PID and can't produce that error message.

**Reply by jkblm** [reply]
Thank you very much :). I will start using it this week to capture USB commands to my RF switch and make a custom keyboard to control each RF switch.

**Comment by pljones** [comment]
Amazon have a couple of {Insert Random Company Name Here}-branded suppliers selling these keyworded to match your board.
https://www.amazon.co.uk/dp/B0GCP9BSHW (https://www.amazon.co.uk/s?k=Histeaa is quite fun...)

<img width="68" height="35" alt="image" src="https://github.com/user-attachments/assets/b95d4701-42f5-41bb-b87d-5cd58ae7fae3" />

I'm guessing that "4" on the 4TG100C is the speed rating?  Meaning it's not a 5 or 6, per the project instructions (LCMXO2-2000HC-5TG100 replacement from the BOM) and means it's definitely not worth the £40, right?

**Reply by ataradov** [reply]
Hard to tell if it is worth it. It is certainly does not match FPGA tool speed report, but the consensus seems to be that it works fine and I have not seen a single report of issues that could be traced to insufficient FPGA grade. I'd say at this point it is safe to assume that grade 4 works fine at least at room temperatures. 

It is safer to get correct grade FPGA, but I would not stress a lot if grade 4 is all you can get.

**Reply by pljones** [reply]
OK.  It'd only be "not worth it" if it wouldn't work at all.  If it's been reported working, great.  I found only https://github.com/ataradov/usb-sniffer/discussions/57 which didn't say "Yay!  Got it working with a -4".

(It's just annoying they don't spent the extra couple of $$ on the -5 or -6 for a reliable kit -- I don't see the point in scraping the floor of the spec to make that tiny margin profit when they could just increase the price to cover it: people who'll buy one will still buy one and more likely if it's the right chip...)
